airbyte-commons/src/main/java/io/airbyte/commons/version/AirbyteProtocolVersionRange.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.commons.version;
+
+public class AirbyteProtocolVersionRange {
```

Comment:
Also the `Version` class itself already has methods that handle comparison to other `Version` objects (`compatibleVersionCompareTo`, `greaterThan`, `lessThan`, etc.), should we be using one of those method here?

Reply:
Not feeling super strong either way about using a record.

About the `Version` comparison, none of the current behaviors matches which is what led me to add this `AirbyteProtocolVersionRange` in the first place.
* `compatibleVersionCompareTo` is closer to an exact compare that check if the major, minor matches and ignores the patch.
* `greaterThan` is doing a Major.minor check, in the protocol version version case, we are only concerned about major because minors should be backward compatible within a major by design.

Reply:
Rewrote the class as a record.